android/app/src/main/java/au/com/robin/sms/SmsSend.kt

```diff
@@ -0,0 +1,47 @@
+package au.com.robin.sms
+
+import android.Manifest
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.telephony.SmsManager
+import android.telephony.SubscriptionManager
+import android.util.Log
+import androidx.annotation.RequiresPermission
+
+class SmsSend {
+
+    companion object {
+        private const val LOG_TAG: String = "SmsSend"
+        @RequiresPermission(Manifest.permission.READ_PHONE_STATE)
+        fun getSmsManager(context: Context, intent: Intent): SmsManager? {
+            val slot = intent.getIntExtra("slot", -1)
+
+            if (slot == -1) {
```

Comment:
Comment about all these different conditions of getting the SmsManager.

Reply:
Yes, thank you for pointing that out!